spec/models/king_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe King, type: :model do
+	let(:king) {create :king}
+
+  describe "King's valid_move" do
+  	it "should be inside destination, blocked" do
+  		expect(king.valid_move?(1, 6)).to eq(false)
+  		expect(king.valid_move?(1, 4)).to eq(false)
+  		expect(king.valid_move?(2, 5)).to eq(false)
+  	end
+
+    it "shouldn't be outside destination" do
+      expect(king.valid_move?(1, 8)).to eq(false)
+      expect(king.valid_move?(1, 2)).to eq(false)
+      expect(king.valid_move?(3, 8)).to eq(false)
+    end
```

Comment:
The first two expectations here won't be met just because the move is obstructed, so if the goal with this block is to test the custom logic outside of `is_move_obstructed?` and `is_move_malformed?`, you should first make sure that the board and the move are set up in such a way that both of those will return `false`.